datastore/prop.go

```diff
@@ -183,7 +183,7 @@ func validateChildType(t reflect.Type, fieldName string, flatten, prevSlice bool
 
 	switch t.Kind() {
 	case reflect.Slice:
-		if flatten && prevSlice {
+		if flatten && prevSlice && t.Elem().Kind() != reflect.Uint8 {
```

Comment:
Maybe a dumb question: why `reflect.Uint8`? How does a slice of strings get caught here?

Reply:
byte is an alias of Uint8. 
For slice of strings, `t.Elem().Kind() != reflect.Uint8` will be false since `t.Elem().Kind()` will return String

Reply:
Got it. In that case a little comment to that effect would be nice for future readers. Something like, "Uint8 is an alias for byte. A slice of bytes is acceptable. A slice of others types is not."